File: jupyter2/ps/iteration/gaussian_integral/jl/gaussian_integral.jl
### begin my answer

### begin hidden
function gaussian_integral(a, n)
    dx = 2a / n
    s = 0.0
    for i in 0:n-1
        x = -a + i * dx
        s += exp(-(x * x)) * dx
    end
    s
end
### end hidden
### end my answer

function main()
    @assert abs(gaussian_integral(1, 1000) - 1.493648) < 1.0e-6
    @assert abs(gaussian_integral(2, 2000) - 1.764163) < 1.0e-6
    @assert abs(gaussian_integral(10, 10000) - sqrt(pi)) < 1.0e-6
    println("OK")
end

main()

File: jupyter2/ps/iteration/gaussian_integral/rs/gaussian_integral.rs
/** begin my answer */

/** begin hidden */
fn gaussian_integral(a: f64, n: i64) -> f64 {
    let dx = 2.0 * a / (n as f64);
    let mut sum = 0.0;
    for i in 0..n {
        let x = -a + (i as f64) * dx;
        sum += (-x * x).exp() * dx;
    }
    sum
}
/** end hidden */
/** end my answer */

fn main() {
    let pi = std::f64::consts::PI;
    assert!((gaussian_integral(1.0, 1000) - 1.493648).abs() < 1.0e-6);
    assert!((gaussian_integral(2.0, 2000) - 1.764163).abs() < 1.0e-6);
    assert!((gaussian_integral(10.0, 10000) - pi.sqrt()).abs() < 1.0e-6);
    println!("OK");
}
